File: elementary_computer_science/VTT_THPT/height.cpp
#include <iostream>
#define N int(1e6)
using namespace std;
int p[N+3];
void sieve() {
	fill(p + 2, p + N + 1, 1);
	for (int i = 2; i*i <= N; ++i)
		if (p[i] == 1)
			for (int j = i*i; j <= N; j += i) p[j] = 0;
}

int sum_digit(int x) {
	int s = 0;
	while (x > 0) {
		 s += x % 10;
		 x /= 10;
	}
	return s;
}

int main() {
	sieve();
	int h, n, cnt = 0;
	cin >> n >> h;
	for (int i = 2; i <= n; ++i)
		if (p[i] == 1 && sum_digit(i) == h) {
			cout << i << '\n';
			++cnt;
		}
	cout << cnt;
}

File: elementary_computer_science/VTT_THPT/height.py
N = int(1e6)
def sieve():
	p[0] = p[1] = 0
	for i in range(2, int(N**.5) + 1):
		if (p[i] == 1):
			for j in range(i*i, N, i):
				p[j] = 0
def sum_digit(x):
	s = 0
	while (x > 0):
		s += x % 10
		x //= 10
	return s
p = [1]*(N + 3)
sieve()
n, h = map(int, input().split())
cnt = 0
for i in range(2, n + 1):
	if (p[i] == 1 and sum_digit(i) == h):
		print(i)
		cnt += 1
print(cnt)